### ISSUE TITLE
Move set Contact/RawContact Options functions to be part of Insert and Update APIs

### ISSUE THREAD

**Issue by vestrel00** [vestrel00/contacts-android#120]
This should be worked on only after completing #268.

## Problem

The extensions provided in `contacts.core.util.ContactOptions` and `contacts.core.util.RawContactOptions` allows for setting `ContactEntity` and `RawContactEntity` `Options`.

Having a separate function to set the **immutable** `Options` property of `ContactEntity` and `RawContactEntity` may cause confusion and easily lead to things being out of sync. Furthermore, it is currently impossible to update Options along with the other data.

Consider this scenario, which I encountered and had to fix in the sample app;

1. Fetch Contact.
2. Show MutableContact in UI.
3. Update Options (e.g. star/unstar) using the extension functions.
    - This update is immediate, which means that options are immediately updated instead of when the user taps "save".
4. The MutableContact in memory does not contain the updated Options.
    - The MutableContact must be re-fetched in order to get the updated options.

## Solution

- [x] Remove all extensions provided in `contacts.core.util.ContactOptions` and `contacts.core.util.RawContactOptions`
- [x] Convert the **immutable** Options in MutableContact and MutableRawContact to be **mutable**

Apply the required changes to the following insert and update APIs;

- [x] `Insert`
- [x] `ProfileInsert`
- [x] `Update`
- [x] `ProfileUpdate`

Additionally;

- [x] Remove support for setting starred/favorited via group membership? If so, create a separate issue for this breaking change.
- [x] Update sample app
- [x] Update cheatsheet
- [x] Update relevant documentation in code
- [x] Update relevant documentation in docs
- [x] Verify that there are no regressions for related API functions
- [ ] Provide a migration guide as a comment in this issue so that it can be copy-pasted into the release notes

**Comment by vestrel00** [comment]
## Migration guide (to be included in release notes)

[Previously](https://github.com/vestrel00/contacts-android/blob/0.2.4/docs/other/get-set-clear-contact-raw-contact-options.md), to get/set Contact/RawContact options, you would use the extension functions provided in `contacts.core.util.ContactOptions` and `contacts.core.util.RawContactOptions`,

```kotlin
val contactOptions = contact.options(contactsApi)
val rawContactOptions = rawContact.options(contactsApi)

contact.setOptions(contactsApi, contactOptions.mutableCopy{ ... })
rawContact.options(contactsApi, rawContactOptions{ ... })
```

The above extension functions read/write directly from the database and blocked the UI thread. Those functions no longer exist.

Now, you can directly get/set options through the Contact/RawContact entities and read/write APIs without needing any extensions;

```kotlin
val contactOptions = contact.options // yes, this also existed prior to this version
val rawContactOptions = rawContact.options

Contacts(context)
    .update()
    .contacts(
        contact.mutableCopy {
            setOptions { ... }
        }
    )
    .rawContacts(
        rawContact.mutableCopy {
            setOptions { ... }
        }
    )
    .commit()
```

You are now also able to set the options of a new RawContact for insert APIs,

```kotlin
Contacts(context)
    .insert()
    .rawContact {
        setOptions { ... }
    }
    .commit()
```

Read the new documentation for the full guide; https://vestrel00.github.io/contacts-android/other/get-set-clear-contact-raw-contact-options/

**Comment by vestrel00** [comment]
This is included in [0.3.0](https://github.com/vestrel00/contacts-android/releases/tag/0.3.0)!
